Duplicati/Library/Main/Operation/Backup/BackendUploader.cs

```diff
@@ -118,80 +115,55 @@ public Task Run()
                 var max_pending = m_options.AsynchronousUploadLimit == 0 ? long.MaxValue : m_options.AsynchronousUploadLimit;
-                var noIndexFiles = m_options.IndexfilePolicy == Options.IndexFileStrategy.None;
-                var active = 0;
                 var lastSize = -1L;
-                
-                while(!await self.Input.IsRetiredAsync && await m_taskreader.ProgressAsync)
+
+                while (!await self.Input.IsRetiredAsync && await m_taskreader.ProgressAsync)
                 {
                     try
                     {
                         var req = await self.Input.ReadAsync();
 
                         if (!await m_taskreader.ProgressAsync)
                             continue;
-                        
-                        var task = default(KeyValuePair<int, Task>);
-                        if (req is VolumeUploadRequest)
+
+                        if (req is VolumeUploadRequest volumeUpload)
                         {
-                            lastSize = ((VolumeUploadRequest)req).BlockVolume.SourceSize;
-                            if(((VolumeUploadRequest)req).IndexVolume == null)
-                                task = new KeyValuePair<int, Task>(1, UploadFileAsync(((VolumeUploadRequest)req).BlockEntry, null));
+                            lastSize = volumeUpload.BlockVolume.SourceSize;
+                            if (volumeUpload.IndexVolume == null)
+                                inProgress.Add(UploadFileAsync(volumeUpload.BlockEntry));
                             else
-                                task = new KeyValuePair<int, Task>(2, UploadBlockAndIndexAsync((VolumeUploadRequest)req));
+                                inProgress.Add(UploadBlockAndIndexAsync(volumeUpload));
                         }
-                        else if (req is FilesetUploadRequest)
-                            task = new KeyValuePair<int, Task>(1, UploadVolumeWriter(((FilesetUploadRequest)req).Fileset));
-                        else if (req is IndexVolumeUploadRequest)
-                            task = new KeyValuePair<int, Task>(1, UploadVolumeWriter(((IndexVolumeUploadRequest)req).IndexVolume));
-                        else if (req is FlushRequest)
+                        else if (req is FilesetUploadRequest filesetUpload)
+                            inProgress.Add(UploadVolumeWriter(filesetUpload.Fileset));
+                        else if (req is IndexVolumeUploadRequest indexUpload)
+                            inProgress.Add(UploadVolumeWriter(indexUpload.IndexVolume));
+                        else if (req is FlushRequest flush)
                         {
                             try
                             {
-                                while(inProgress.Count > 0)
-                                    await inProgress.Dequeue().Value;
-                                active = 0;
+                                await Task.WhenAll(inProgress);

```

Comment:
I'm not sure if I'm reading this correctly, but it seems that previously if a queued `Task` threw an exception, the remaining `Tasks` would not be awaited.  Now, all the `Tasks` are run until they complete.  If that's true, what are the side effects (if any) or differences in behavior (good or bad)?